test/services/LibraryService.test.ts

```diff
@@ -86,6 +109,95 @@ describe('LibraryService', () => {
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(1);
+          expect(response.body.entry).toEqual(expect.arrayContaining([{ resource: LIBRARY_WITH_NO_DEPS }]));
+        });
+    });
+
+    it('returns a Bundle including the Library, and when the Library has no dependencies and id passed through body', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({ resourceType: 'Parameters', parameter: [{ name: 'id', valueString: 'testLibraryWithNoDeps' }] })
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(1);
+          expect(response.body.entry).toEqual(expect.arrayContaining([{ resource: LIBRARY_WITH_NO_DEPS }]));
+        });
+    });
+
+    it('returns a Bundle including the Library, and when the Library has dependencies and id passed through args', async () => {
+      await supertest(server.app)
+        .get('/4_0_1/Library/testLibraryWithDeps/$package')
+        .expect(200)
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(2);
+          expect(response.body.entry).toEqual(
+            expect.arrayContaining([{ resource: LIBRARY_WITH_DEPS }, { resource: LIBRARY_WITH_NO_DEPS }])
+          );
+        });
+    });
+
+    it('returns a Bundle including the Library, and when the Library has dependencies and id passed through body', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({ resourceType: 'Parameters', parameter: [{ name: 'id', valueString: 'testLibraryWithDeps' }] })
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Bundle');
+          expect(response.body.entry).toHaveLength(2);
+          expect(response.body.entry).toEqual(
+            expect.arrayContaining([{ resource: LIBRARY_WITH_DEPS }, { resource: LIBRARY_WITH_NO_DEPS }])
+          );
+        });
+    });
+
+    it('throws a 400 error when no url or id included in request', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library/$package')
+        .send({ resourceType: 'Parameters', parameter: [] })
+        .set('content-type', 'application/fhir+json')
+        .expect(400)
+        .then(response => {
+          expect(response.body.issue[0].code).toEqual('BadRequest');
+          expect(response.body.issue[0].details.text).toEqual(
+            'Must provide identifying information via either id or url parameters'
+          );
+        });
+    });
+
+    it('throws a 404 error when no library matching id and url can be found', async () => {
```

Comment:
I don't think we have a Measure/$package test with both id and url specified but one is invalid and the other is valid, so if we add a test for that here then we should also do the same for Measure/$package